cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/cdap/CDAPLoad.java

```diff
@@ -101,20 +98,18 @@ public void collect() throws IOException {
     // reset all metrics
     reset();
 
-    long currentTimeSecs = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+    int currentTimeSecs = (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
     // We want metrics in the last hour
-    long startTimeSecs = currentTimeSecs - (60 * 60);
+    int startTimeSecs = currentTimeSecs - (60 * 60);
     // We want to aggregate metrics in a sliding window of the past hour from the time the request is made.
     // To do this, query metrics with the minute resolution, so you get 60 points, then aggregate on the client side.
     // Not using hourly resolution here, because that wouldn't give aggregated metrics for the past hour, but
     // just aggregated metrics for the current hour (if called 15 mins past the hour, it will only give an aggregated
     // value for the past 15 mins).
-    Collection<MetricTimeSeries> metricTimeSeries = metricStore.query(
-      // TODO: CDAP-8207 Have to use this constructor currently, because the default limit does not work
```

Comment:
I don't really know what this TODO means. But is it resolved by your change?

Reply:
I updated the `DirectMetricsSystemClient` to use the same constructor. Also, for the `RemoteMetricsSystemClient` case, it is hitting the REST API and apparently it is not affected by this bug.